mlflow/utils/search_utils.py

```diff
@@ -1129,6 +1134,9 @@ def _does_model_version_match_clauses(cls, mv, sed):  # pylint: disable=argument
     @classmethod
     def filter(cls, model_versions, filter_string):  # pylint: disable=arguments-renamed
         """Filters a set of model versions based on a search filter string."""
+        model_versions = filter(
+            lambda mv: mv.current_stage != STAGE_DELETED_INTERNAL, model_versions
+        )
```

Comment:
```suggestion
        model_versions = [
            for mv in model_versions if mv.current_stage != STAGE_DELETED_INTERNAL
        ]
```

to avoid returning a filter object on line 1141.